### PR TITLE
Release beta-6

### PR DESCRIPTION
**Pull request by descorp**
# Open PR

## Features

* add standardised error codes

### iOS

* add ApplePay component
* add Swift support to **RedirectComponentProxy**

### Android

* add GooglePay component
* add `onFail` to DropIn when shopper canceled 

## Fixes

* type comparison lowcased #39 
* add clean effect to AdyenCheckoutContext #44 
* Android: add loading view to redirect action to prevent "closed browser" UX pitfall #46 

### Android 

* static `handle` method in ActionHandler

## Other

* Improve Example app structure
* Update README
* Change project logo

### CODE REVIEW

**Review by mohammedDehairy — DISMISSED** [review]

File: ios/Components/AdyenDropIn.swift [hunk 1]
```diff
@@ -55,36 +46,28 @@ final internal class AdyenDropIn: BaseModule {
 
         if let payment = parser.payment {
             config.payment = payment
-
-            // Apple Pay
-            if let applepayConfig = ApplepayConfigurationParser(configuration: configuration).tryConfiguration(amount: payment.amount) {
-                config.applePay = applepayConfig
+            (try? ApplepayConfigurationParser(configuration: configuration).buildConfiguration(amount: payment.amount)).map {
+                config.applePay = $0
```
[the remark this answers is not in the document]
Comment (descorp) [reply ios/Components/AdyenDropIn.swift:50]: 😎

File: android/src/main/java/com/adyenreactnativesdk/util/ReactNativeError.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import com.facebook.react.bridge.ReadableMap
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.bridge.WritableNativeMap
+import java.lang.Exception
+
+object ReactNativeError {
+
+    private const val MESSAGE_KEY = "message"
+    private const val REASON_KEY = "reason"
+    private const val DISCRIPTION_KEY = "discription"
```
Comment (jreij) [inline android/src/main/java/com/adyenreactnativesdk/util/ReactNativeError.kt:21]: Typo here

**Review by jreij — DISMISSED** [review]

File: android/src/main/java/com/adyenreactnativesdk/util/ReactNativeError.kt (hunk 2 above)
Comment (descorp) [reply android/src/main/java/com/adyenreactnativesdk/util/ReactNativeError.kt]: Fixed in [b9edbc9](https://github.com/Adyen/adyen-react-native/pull/59/commits/b9edbc920769ed44b5d970ba3580c70b170480e9)

**Review by jreij — APPROVED** [review]

File: example/src/CheckoutView.js [hunk 3]
```diff
@@ -1,19 +1,10 @@
...
-  Text,
-  View,
-  Alert,
-} from 'react-native';
-import { Colors } from 'react-native/Libraries/NewAppScreen';
-import { PaymentMethodsContext } from './PaymentMethodsProvider';
+import { SafeAreaView, StyleSheet, Text, View, Alert } from 'react-native';
+import { usePaymentMethods } from './PaymentMethodsProvider';
 import PaymentMethods from './PaymentMethodsView';
+import { ERROR_CODE_CANCELED } from '../../src';
```
Comment (m1aw) [inline example/src/CheckoutView.js:7]: Shouldn't the example just use the lib? It seems to be importing across the project, is this correct?

Comment (descorp) [reply example/src/CheckoutView.js:7]: This is indeed not correct 🤔
But there is no bug, just weird referencing from VSCode.
Following works as well:
```js
import { ERROR_CODE_CANCELED } from '@adyen/react-native';
```

**Pull request merged**